packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -230,8 +235,8 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     }
 
     final int lastIndex = indexOf(lastChild);
-    final double leadingScrollOffset = indexToLayoutOffset(itemExtent, firstIndex);
-    final double trailingScrollOffset = indexToLayoutOffset(itemExtent, lastIndex + 1);
+    final double leadingScrollOffset = firstIndex == 0 ? 0 : indexToLayoutOffset(itemExtent, firstIndex - 1) + itemExtent;
+    final double trailingScrollOffset = indexToLayoutOffset(itemExtent, lastIndex) + itemExtent;
```

Comment:
I don't understand the purpose of this change and the line above.

Reply:
Add trailing padding back to trailingScrollOffset, it should make more sense now.